ext/reflection/php_reflection.c

```diff
@@ -2104,7 +2104,145 @@ ZEND_METHOD(ReflectionFunctionAbstract, getNumberOfRequiredParameters)
 }
 /* }}} */
 
-/* {{{ Returns an array of parameter objects for this function */
+/* {{{ Returns whether a parameter exists or not */
+ZEND_METHOD(ReflectionFunctionAbstract, hasParameter)
+{
+    reflection_object *intern;
+    zend_function *fptr;
+    zend_string *arg_name = NULL;
+    zend_long position;
+    struct _zend_arg_info *arg_info;
+    uint32_t num_args;
+
+    ZEND_PARSE_PARAMETERS_START(1, 1)
+        Z_PARAM_STR_OR_LONG(arg_name, position)
+    ZEND_PARSE_PARAMETERS_END();
+
+    GET_REFLECTION_OBJECT_PTR(fptr);
+
+    num_args = fptr->common.num_args;
+    arg_info = fptr->common.arg_info;
+
+    if (fptr->common.fn_flags & ZEND_ACC_VARIADIC) {
+        num_args++;
+    }
+
+    if (!num_args) {
+        RETURN_FALSE;
+    }
+
+    if (arg_name != NULL) {
+        uint32_t i;
+        position = -1;
+
+        for (i = 0; i < num_args; i++) {
+            for (i = 0; i < num_args; i++) {
+                if (has_internal_arg_info(fptr)) {
+                    if (arg_info[i].name) {
+                        if (strcmp(((zend_internal_arg_info*)arg_info)[i].name, ZSTR_VAL(arg_name)) == 0) {
+                            RETURN_TRUE;
+                        }
+                    }
+                } else {
+                    if (arg_info[i].name) {
+                        if (zend_string_equals(arg_name, arg_info[i].name)) {
+                            RETURN_TRUE;
+                        }
+                    }
+                }
+            }
+        }
+
+        RETURN_FALSE;
+    } else {
+        if (position < 0 || position >= num_args) {
+            RETURN_FALSE;
+        }
+
+        RETURN_TRUE;
+    }
```

Comment:
Add a ValueError for negative positions, as those are a programing error.

Also could just use ``RETURN_BOOL(condition)``